File: java/test03.java
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

public class HelloWorld {

	private static String parameters = 
			"{\"request\":"
			+ "	{\"intent\":"
			+ "		{\"slots\":"
			+ "			{\"Text\":"
			+ "				{\"value\":\"World\"}"
			+ "			}"
			+ "		}"
			+ "	},"
			+ "\"version\": \"local-version\""
			+ "}";
	
	public static void main(String[] args) {
		JsonObject arg = new JsonParser().parse(parameters).getAsJsonObject();
		JsonObject response = HelloWorld.main(arg);
		System.out.println(response);
	}
	
	public static JsonObject main(JsonObject args) {
		    
        JsonObject response = new JsonObject();
        String name = args.getAsJsonObject("request").getAsJsonObject("intent")
				.getAsJsonObject("slots").getAsJsonObject("Text").getAsJsonPrimitive("value").getAsString();
        String speechText = "Hello " + name + ". Welcome to the Openwhisk program";

        response.addProperty("version", args.getAsJsonPrimitive("version").getAsString());
        
        JsonObject resp = new JsonObject();
        JsonObject outputSpeech = new JsonObject();
        outputSpeech.addProperty("type", "PlainText");
        outputSpeech.addProperty("text", speechText);
        resp.add("outputSpeech", outputSpeech);
        
        JsonObject card = new JsonObject();
        card.addProperty("content", speechText);
        card.addProperty("title", "Session");
        card.addProperty("type", "Simple");
        resp.add("card", card);
        
        JsonObject rePrompt = new JsonObject();
        rePrompt.add("outputSpeech", outputSpeech);
        resp.add("reprompt", rePrompt);
        
        resp.addProperty("shouldEndSession", false);
        
        response.add("response", resp);
        response.add("sessionAttributes", new JsonObject());
        return response;
    }

}

File: nodejs/test03.js
function main(params) {
	var replyText = "Please say the phrase that you want me to repeat.";
	if (params.request.type != "LaunchRequest") {
		var intent = params.request.intent;
		var sessionAttributes = params.session.attributes;
		
		if (intent.name == "RawText")
			replyText = intent.slots.Text.value.toLowerCase().replace(/\W\s/g, '');
	}
	
	console.log("replyText : " + replyText);

	var outputSpeech = {
		"type": "SSML",
		"ssml": "<speak>" + replyText + "</speak>"
	};

	return {
		"version": "1.0",
		"response": {
			outputSpeech,
			"reprompt": { outputSpeech },
			"shouldEndSession": false
		}
	}
}

exports.main = main;
